src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,261 @@
+        {
+            // Register and Execute are not concurrency-safe.
+            // Avoiding calling Register while Execute is running is the caller's responsibility.
+            ThrowIfDisposed();
+
+            // Register all in logPolling, there is no problem if we get 2 notifications of the new blob
+            Task registerLogPolling = _pollLogStrategy.RegisterAsync(container, triggerExecutor, cancellationToken);
+
+            ICollection<ITriggerExecutor<IStorageBlob>> containerRegistrations;
+            if (!_registrations.TryGetValue(container, out containerRegistrations))
+            {
+                containerRegistrations = new List<ITriggerExecutor<IStorageBlob>>();
+                _registrations.Add(container, containerRegistrations);
+            }
+
+            containerRegistrations.Add(triggerExecutor);
+
+            if (!_lastSweepCycleStartTime.ContainsKey(container))
+            {
+                _lastSweepCycleStartTime.Add(container, DateTime.MinValue);
+            }
+
+            if (!_currentSweepCycleStartTime.ContainsKey(container))
+            {
+                _currentSweepCycleStartTime.Add(container, DateTime.MinValue);
+            }
+
+            if (!_continuationTokens.ContainsKey(container))
+            {
+                _continuationTokens.Add(container, null);
+            }
+            await registerLogPolling;
+        }
+
+        public async Task<TaskSeriesCommandResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+
+            Task LogPollingTask = _pollLogStrategy.ExecuteAsync(cancellationToken);
+            List<IStorageBlob> failedNotifications = new List<IStorageBlob>();
+            List<Task> notifications = new List<Task>();
+
+            // Drain the background queue of blob written notifications.
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                IStorageBlob blob;
+
+                if (!_blobsFoundFromScanOrNotification.TryDequeue(out blob))
+                {
+                    break;
+                }
+
+                notifications.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
+            }
+            Task.WhenAll(notifications).Wait();
+
+            List<Task> pollingTasks = new List<Task>();
+            foreach (IStorageBlobContainer container in _registrations.Keys)
+            {
+                pollingTasks.Add(PollAndNotify(container, cancellationToken, failedNotifications));
+            }
+
+            // Re-add any failed notifications for the next iteration.
+            foreach (IStorageBlob failedNotification in failedNotifications)
+            {
+                _blobsFoundFromScanOrNotification.Enqueue(failedNotification);
+            }
+
+            await LogPollingTask;
```

Comment:
This is currently serialized with the wait on the next line. 
You could do:
pollingTasks.Add(LogPollingTask) and then the waits will happen together.